docs/concepts/select.md

```diff
@@ -353,6 +353,61 @@ export class CityService {
 
 now we can use this `zooThemeParks` selector anywhere in our application.
 
+### Inheriting selectors  
+
+When we have states that share similar structure, we can extract the shared selectors into a base class which we can later extend from. If we have an `entities` field on multiple states, we can create a base class containing a dynamic `@Selector()` for that field, and extend from it on the `@State` classes like this.
+
+```TS
+export class EntitiesState {
+
+  static entities() {
+    return createSelector([this], (state: { entities: any[] }) => {
+      return state.entities;
+    });
+  }
+
+  //...
+
+}
+```
+
+And extend the `EntitiesState` class on each `@State` like this:
+
+```TS
+
+@State<string[]>({
+  name: 'animals',
+  defaults: []
+})
+export class ZooState extends EntitiesState {  
```

Comment:
This would need to be a state that has an `{ entities: any[] }` model type.
I would recommend that we don't redefine what a ZooState is for consistency.
Maybe we could have UsersState and RolesState as entity states?

Reply:
ok, i finally went with UsersState and ProductsState, i also added the interfaces for UsersStateModel and ProductsStateModel and update the `name` in the `@State` options